diff --git a/modin/numpy/arr.py b/modin/numpy/arr.py
index 42f79f0a..dffc1188 100644
--- a/modin/numpy/arr.py
+++ b/modin/numpy/arr.py
@@ -2239,7 +2239,7 @@ class array(object):
         na_mask = self._query_compiler.isna().any(axis=apply_axis)
         if na_mask.any(axis=apply_axis ^ 1).to_numpy()[0, 0]:
             na_idxs = self._query_compiler.isna().idxmax(axis=apply_axis)
-            result = na_mask.where(na_idxs, result)
+            result = na_idxs.where(na_mask, result)
         new_ndim = self._ndim - 1 if not keepdims else self._ndim
         if new_ndim == 0:
             result = result.to_numpy()[0, 0]
@@ -2304,7 +2304,7 @@ class array(object):
         na_mask = self._query_compiler.isna().any(axis=apply_axis)
         if na_mask.any(axis=apply_axis ^ 1).to_numpy()[0, 0]:
             na_idxs = self._query_compiler.isna().idxmax(axis=apply_axis)
-            result = na_mask.where(na_idxs, result)
+            result = na_idxs.where(na_mask, result)
         new_ndim = self._ndim - 1 if not keepdims else self._ndim
         if new_ndim == 0:
             result = result.to_numpy()[0, 0]
diff --git a/modin/numpy/test/test_array_math.py b/modin/numpy/test/test_array_math.py
index b142200f..d419ede2 100644
--- a/modin/numpy/test/test_array_math.py
+++ b/modin/numpy/test/test_array_math.py
@@ -34,6 +34,15 @@ def test_argmax_argmin(data, op):
     assert_scalar_or_array_equal(modin_result, numpy_result)
 
 
+@pytest.mark.parametrize("axis", [0, 1])
+@pytest.mark.parametrize("op", ["argmin", "argmax"])
+def test_argmax_argmin_2d_with_nan(axis, op):
+    data = numpy.array([[1.0, 2.0, 3.0], [4.0, 5.0, numpy.nan]])
+    numpy_result = getattr(numpy, op)(data, axis=axis)
+    modin_result = getattr(np, op)(np.array(data), axis=axis)
+    assert_scalar_or_array_equal(modin_result, numpy_result)
+
+
 def test_rem_mod():
     """Tests remainder and mod, which, unlike the C/matlab equivalents, are identical in numpy."""
     a = numpy.array([[2, -1], [10, -3]])
